authlib/jose/rfc7516/jwe.py

```diff
@@ -197,6 +338,110 @@ def deserialize_compact(self, s, key, decode=None, sender_key=None):
             payload = decode(payload)
         return {'header': protected, 'payload': payload}
 
+    def deserialize_json(self, obj, key, decode=None, sender_key=None):
+        obj = ensure_dict(obj, 'JWE')
+
+        if 'protected' in obj:
+            protected = extract_header(to_bytes(obj['protected']), DecodeError)
+        else:
+            protected = None
+
+        unprotected = obj.get('unprotected')
+
+        recipients = obj['recipients']
+        for recipient in recipients:
+            recipient['encrypted_key'] = extract_segment(
+                to_bytes(recipient['encrypted_key']), DecodeError, 'encrypted key')
+
+        if 'aad' in obj:
+            jwe_aad = extract_segment(to_bytes(obj['aad']), DecodeError, 'JWE AAD')
+        else:
+            jwe_aad = None
+
+        iv = extract_segment(to_bytes(obj['iv']), DecodeError, 'initialization vector')
+
+        ciphertext = extract_segment(to_bytes(obj['ciphertext']), DecodeError, 'ciphertext')
+
+        tag = extract_segment(to_bytes(obj['tag']), DecodeError, 'authentication tag')
+
+        shared_header = JWESharedHeader(protected, unprotected)
+
+        alg = self.get_header_alg(shared_header)
+        enc = self.get_header_enc(shared_header)
+        zip_alg = self.get_header_zip(shared_header)
+
+        self._validate_sender_key(sender_key, alg)
```

Comment:
Do we validate recipient keys and check that sender and recipient keys are on the same curve?

Reply:
This is done by the underlying crypto library.